Connected System Template (CST) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/helloworld/templates/GoogleJsonResponseExceptionHandler.java

```diff
@@ -0,0 +1,35 @@
+package com.mycorp.helloworld.templates;
+
+import com.appiancorp.connectedsystems.templateframework.sdk.IntegrationError;
+import com.appiancorp.connectedsystems.templateframework.sdk.IntegrationResponse;
+import com.appiancorp.connectedsystems.templateframework.sdk.diagnostics.IntegrationDesignerDiagnostic;
+import com.appiancorp.connectedsystems.templateframework.sdk.oauth.ExpiredTokenException;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+
+public class GoogleJsonResponseExceptionHandler {
+  public static IntegrationResponse handleException(
+      GoogleJsonResponseException e,
+      IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder,
+      DiagnosticsFactory diagnosticsFactory,
+      long startTime) {
+    if (e.getStatusCode() == 401) {
+      //Google returns a 401 exception if your credential is not authorized or expired. Throw a
+      //ExpiredTokenException when this happens, Appian will try to refresh the token for up to
```

Comment:
Appian will attempt to refresh your token. 3 times is just our current policy and is subject to change